.github/actions/bump-version/action.js

```diff
@@ -0,0 +1,38 @@
+const core = require('./core');
+
+function bumpVersion(currentVersion, bumpType, prerelease) {
+  let newVersion = calculateNewVersion(currentVersion, bumpType);
+
+  if (prerelease !== undefined && prerelease !== '') {
+    newVersion = `${newVersion}.${prerelease}`;
+  }
+  core.setOutput('previous_version', currentVersion)
+  core.setOutput('previous_version_tag', `v${currentVersion}`)
+  core.setOutput('version', newVersion);
+  core.setOutput('version_tag', `v${newVersion}`)
+
+  return newVersion;
+}
+
+function calculateNewVersion(currentVersion, bumpType) {
+  const [major, minor, patch] = currentVersion.split('.');
+  let newVersion;
+
+  switch (bumpType) {
```

Comment:
I know you said you looked at a few actions (I remember having the same struggle lol), but I eventually got https://github.com/mathieudutour/github-tag-action to work for our use case in the dashboard repo -- if you use that with dry_run, does that work for you here?

Reply:
That looks like a nice action overall, but it presumes you are trying to use [semantic release](https://github.com/semantic-release/semantic-release) commit message conventions. Semantic release is a nice idea we should consider adopting in our open source projects in the future but is not something we are doing currently. This action lets you specify the "default bump" which would give the behavior I want now, but I don't really want this action to make a different determination about release version than the one I ask for somewhere down the road because of how somebody might format a commit message.